apps/tenant-management-webapp/src/app/lib/autoComplete.ts

```diff
@@ -198,3 +199,21 @@ const getElementSuggestion = (element) => {
   }
   return suggest;
 };
+const getPositionX = (text: string, search: string): number => {
+  const textArr = text.split('\n');
+  for (let i = textArr.length - 1; i >= 0; i--) {
+    if (textArr[i].indexOf(search) > -1) {
+      return i + 1;
+    }
+  }
+  return 10000;
+};
+export const triggerInScope = (text: string, position: Position) => {
+  if (text.indexOf('{{') !== -1) {
+    if (getPositionX(text, '{{') <= position.lineNumber && getPositionX(text, '{{') >= position.lineNumber) {
+      return true;
+    }
+    return false;
+  }
+  return false;
+};
```

Comment:
This would be more succinct and easier to read if you said
return text.indexOf('{{') !== -1 &&
           getPositionX(text, '{{') <= position.lineNumber &&
           getPositionX(...) >= position.lineNumber
           
 In fact, having written it out, did you mean petPosition(text, '}}') >= position.lineNumber?
 Could you write a unit test for this method?

Reply:
New code be committed, please give another review, Thanks